### PR TITLE
66 keyword search

### LINKED ISSUE #66: Keyword search
## Description
As a purchaser I want to be able to search for important keywords because I want to be able to search for related words and have related contracts show up.

## Acceptance criteria
- [x] When the search query contains one or more keywords associated with a contract, that contract is shown in search results
- [x] Only the entirety of keywords are matched, there are no partial matches of keywords
- [x] The tags are stored as some sort of collection in the contract

## Dependencies
- #59 
- #64

### PR DESCRIPTION
**Pull request by martinjonsson01**
## Summary
Adds tag searching to the search engine used by the contracts search box. You can now search for predefined tags and have contracts containing those tags show up.

chrome_BDbqxgmvgz

## Definition of done
- [x] All acceptance criteria are completed.
- [x] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [ ] The code is reviewed and accepted by (at least) two other team members.
- [x] Views adhere to design style.

Closes #66

### CODE REVIEW

File: tests/Application.Tests/Search/Modules/TagSearchModuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        _cut = new TagSearch();
+
+        string[] tags = { "Tag1", "Tag2", "Tag3", };
+        _contract = new Contract { Tags = tags, };
+    }
+
+    [Theory]
+    [InlineData(0)]
+    [InlineData(1)]
+    [InlineData(2)]
```
Comment (JacobBredin) [inline tests/Application.Tests/Search/Modules/TagSearchModuleTests.cs:24]: Fancy 😎

**Review by JacobBredin — APPROVED** [review]
Nicely done 🚀 Found no bugs 🐛

File: .idea/.idea.ContractPortal/.idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="SwUserDefinedSpecifications">
+    <option name="specTypeByUrl">
+      <map />
+    </option>
+  </component>
+</project>
```
Comment (jonathan-lindqvist) [inline .idea/.idea.ContractPortal/.idea/misc.xml:8]: Should this be in the gitignore?

File: src/Infrastructure/Contracts/FakeContractRepository.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿using System.Collections.ObjectModel;
 
 using Application.Contracts;
+
```
Comment (jonathan-lindqvist) [inline src/Infrastructure/Contracts/FakeContractRepository.cs:4]: 🌌👾 🌌👾 🌌👾 🌌👾 🌌👾 🌌👾 🌌👾 🌌👾 🌌👾 🌌

**Review by jonathan-lindqvist — APPROVED** [review]
Sick!

File: .idea/.idea.ContractPortal/.idea/misc.xml (hunk 2 above)
Comment (martinjonsson01) [reply .idea/.idea.ContractPortal/.idea/misc.xml]: Yup

File: src/Infrastructure/Contracts/FakeContractRepository.cs (hunk 3 above)
Comment (martinjonsson01) [reply src/Infrastructure/Contracts/FakeContractRepository.cs:4]: #blametheformatter

**Pull request merged**